File: scala/src/main/scala/forcomprehension.scala
import com.jambit.practicalresult.*

@main def forComprehension: Unit =
  val result: Either[String, List[Int]] =
    for {
      a <- Right(4)
      list1 <- listService(a)
      list2 <- listService(a - 2)
    } yield list1 ++ list2

  println(result)

File: kotlin/src/main/java/ForComprehension.kt
import arrow.core.Either
import arrow.core.computations.either

suspend fun main() {
  val result: Either<String, List<Int>> =
    either {
      val a = Either.Right(4).bind()
      val list1 = listService(a).bind()
      val list2 = listService(a - 2).bind()
      list1 + list2
    }

  println(result)
}